### PR TITLE
hw03 submission (todo: property tests for rbtrees) - Tea Stefanovska

### PR DESCRIPTION
**Pull request by tejchu**

### CODE REVIEW

File: hw03/hw01/test/Test/BankersQueue.hs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+unitTests = [isEmptyTests
+  , enqueueTests
+  , headTests
+  , tailTests]
+
+
+props = [
+    testProperty "prop_enqueue" prop_enqueue,
+    testProperty "prop_tail" prop_tail,
+    testProperty "prop_isEmpty" prop_isEmpty]
```
Comment (kajigor) [inline hw03/hw01/test/Test/BankersQueue.hs]: It's better to use the textual representation of the property as the description of the test.

File: hw03/hw01/test/Test/BankersQueue.hs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+prop_isEmpty :: Property
+prop_isEmpty = property $ do
+  xs <- forAll genIntList
+  let queue = createEmpty :: BankersQueue Int
+  let q = foldl enqueue queue xs
+  let isEmptyResult = isEmpty q
+  if null xs 
+      then assert $ isEmptyResult
+      else assert $ not $ isEmptyResult
```
Comment (kajigor) [inline hw03/hw01/test/Test/BankersQueue.hs]: This code is duplicated for each queue. Can you come up with a way to abstract it away to get rid of the duplication?

**Review by kajigor — CHANGES_REQUESTED** [review]
A lot of unnecessary code duplication which should be got rid of.

File: hw03/hw01/test/Test/BankersQueue.hs (hunk 1 above)
Comment (tejchu) [reply hw03/hw01/test/Test/BankersQueue.hs]: Is the current description okay, or should I make other changes?

File: hw03/hw01/test/Test/BankersQueue.hs (hunk 2 above)
Comment (tejchu) [reply hw03/hw01/test/Test/BankersQueue.hs]: I realized that, I have created a new file only for the property-based tests for the queues. Is this something that satisfies the request or should I do anything differently?

**Comment by kajigor** [comment]
Ok

**Pull request closed without merging**